### PR TITLE
vertex based patch creation

### PR DESCRIPTION
**Pull request by cwsmith**
This PR adds support for creation of small sets of elements (the highest dimension mesh entity, `dim`) surrounding each vertex in the mesh.  The patches are formed by first collecting the upward adjacent elements to each vertex and then successively adding elements to the patch that are adjacent to the current elements via sides (entities of dimension `dim-1`) until the minimum patch size is reached for each vertex.  For use with a partitioned mesh, the mesh partition needs to be created with ghosting enabled (i.e., `mesh.set_parting(OMEGA_H_GHOSTED)`).

Note, Kokkos is required for shared memory parallel segment sorting via team sort.

This functionality is needed to support super convergent patch recovery for the [BER land ice project](https://fanssie.github.io/).

- [x]  add 3d test case
- [x]  add partitioned mesh test case

### CODE REVIEW

File: src/Omega_h_patches.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        auto elm = patch_elms[j];
+        for(int k=adjElms_offsets[elm]; k<adjElms_offsets[elm+1]; k++) {
+          patchExpDup_elms[idx++] = adjElms_elms[k];
+        }
+      }
+    }
+  });
+  Graph patchExpDup(patchExpDup_offsets,patchExpDup_elms);
+  auto sorted = adj_segment_sort(patchExpDup);
+  auto dedup = remove_duplicate_edges(sorted);
```
Comment (jacobmerson) [inline src/Omega_h_patches.cpp:95]: I'm not sure that I fully understand the algorithm here. Why is the deduplication necessary?

**Review by jacobmerson — COMMENTED** [review]
I think it might be helpful to have a couple comments in the `expandPatches` function that explains the steps of the algorithm. 

Is the naming convention you are using functions in the anonymous namespace use underscores and other functions are camel case? That's fine just wasn't positive if there is something else I wasn't grokking.

File: src/Omega_h_patches.cpp (hunk 1 above)
Comment (cwsmith) [reply src/Omega_h_patches.cpp:107]: The description of the patch expansion procedure was just added: https://github.com/SCOREC/omega_h/pull/112/commits/102a0d8f2858b5bfe0b743230d316cf9a4470409.

Good catch on the mix of function name styles - that is fixed with https://github.com/SCOREC/omega_h/pull/112/commits/8d65294858961787b95ef1fc0b3988dc195eb08b.

**Review by jacobmerson — APPROVED** [review]

**Comment by jacobmerson** [comment]
Thanks for adding the docs. I am fine with including this if you are ready.

**Pull request merged**